src/errors/validation_exception.rs

```diff
@@ -201,6 +272,22 @@ impl ValidationError {
         Ok(PyString::new(py, s))
     }
 
+    fn derive(&self, py: Python, errors: &PyList) -> PyResult<Py<Self>> {
+        Py::new(
+            py,
+            Self {
+                exceptions: errors
+                    .iter()
+                    .map(WrappedError::extract)
+                    .collect::<Result<_, _>>()
+                    .unwrap(),
+                message: self.message.clone(),
```

Comment:
again please can we use `clone_ref(py)`, I'm sure the difference is small, but still.